xapian-core/cluster/cluster.cc

```diff
@@ -132,92 +141,357 @@ TermListGroup::get_termfreq(const string &tname) {
 }
 
 doccount
-ClusterSet::num_of_clusters() {
-    LOGCALL(API, doccount, "ClusterSet::num_of_clusters()", "");
+ClusterSet::size() {
+    LOGCALL(API, doccount, "ClusterSet::size()", NO_ARGS);
     return clusters.size();
 }
 
 doccount
 ClusterSet::cluster_size(clusterid cid) {
     LOGCALL(API, doccount, "ClusterSet::cluster_size()", cid);
-    map<clusterid, vector<Document> >::iterator i;
-    i = clusters.find(cid);
-    if (i == clusters.end())
+    unsigned int s = clusters.size();
+    if (cid >= s)
 	throw RangeError("The mentioned clusterid was out of range", 103);
```

Comment:
Yes that was a mistake on my part! When I wrote it, I didn't know where to find what different errorno values would be interpreted as. I hadn't added the errono value, but I got an error saying less arguments, so I just added one. Will change that.